### ISSUE TITLE
Circular object not detected

### ISSUE THREAD

**Issue by Akxe** [kolodny/jsan#32]
I have made as simple stackblitz as possible: https://stackblitz.com/edit/jsan-stringify-error?file=index.ts

This is the key point of it:
```js
const carrier = {
  id: "Carrier 1",
  map: new Map()
};

const period = {
  id: "Period 1",
  carriers: new Set([carrier])
};

carrier.map.set(period, {});

const result = stringify([carrier], undefined, null, true);
```

**Comment by Akxe** [comment]
@kolodny any stand on this issue?

**Comment by kolodny** [comment]
Good catch. Fixed and published to 3.1.14. Thanks!
